core/outbox_test.go

```diff
@@ -0,0 +1,201 @@
+package core_test
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/filecoin-project/go-filecoin/actor"
+	"github.com/filecoin-project/go-filecoin/actor/builtin/account"
+	"github.com/filecoin-project/go-filecoin/actor/builtin/storagemarket"
+	"github.com/filecoin-project/go-filecoin/address"
+	"github.com/filecoin-project/go-filecoin/core"
+	tf "github.com/filecoin-project/go-filecoin/testhelpers/testflags"
+	"github.com/filecoin-project/go-filecoin/types"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOutbox(t *testing.T) {
```

Comment:
These are ported from the `msg.Sender` tests, but with fakes instead of real dependencies (chain store, state HAMT etc).

I dropped a few internal tests for `nextNonce` and added some nonce checks to the existing tests, so this test is now in a `_test` package.